src/Symfony/Component/Form/Extension/Csrf/Type/FormTypeCsrfExtension.php

```diff
@@ -73,6 +73,7 @@ public function finishView(FormView $view, FormInterface $form, array $options):
             $csrfForm = $factory->createNamed($options['csrf_field_name'], HiddenType::class, $data, [
                 'block_prefix' => 'csrf_token',
                 'mapped' => false,
+                'attr' => ['data--csrf-protection' => true],
```

Comment:
A particular reason to use `--` ? As it won't work in dataSet, it could limit other implementations

Reply:
as a kind of namespace to prevent unwanted collisions
getAttributes works universally so not sure dataSet is needed here
no strong opinion either :)

Reply:
I renamed this to `data-csrf-protection` and I also added a `csrf_data_attr` form option to configure that name, this should help escape any collisions.